tensorflow/python/ops/rnn.py

```diff
@@ -463,6 +463,109 @@ def bidirectional_rnn(cell_fw, cell_bw, inputs,
+      but needed for back prop from GPU to CPU.  This allows training RNNs
+      which would typically not fit on a single GPU, with very minimal (or no)
+      performance penalty.
+    time_major: The shape format of the `inputs` and `outputs` Tensors.
+      If true, these `Tensors` must be shaped `[max_time, batch_size, depth]`.
+      If false, these `Tensors` must be shaped `[batch_size, max_time, depth]`.
+      Using `time_major = True` is a bit more efficient because it avoids
+      transposes at the beginning and end of the RNN calculation.  However,
+      most TensorFlow data is batch-major, so by default this function
+      accepts input and emits output in batch-major form.
+    dtype: (optional) The data type for the initial state.  Required if
+      initial_state is not provided.
+    sequence_length: An int32/int64 vector, size `[batch_size]`,
+      containing the actual lengths for each of the sequences.
+      either of the initial states are not provided.
+    scope: VariableScope for the created subgraph; defaults to "BiRNN"
+
+  Returns:
+    A tuple (outputs, output_state_fw, output_state_bw) where:
+      outputs: The RNN output `Tensor`.
+        If time_major == False (default), this will be a `Tensor` shaped:
+          `[batch_size, max_time, cell.output_size]`.
+        If time_major == True, this will be a `Tensor` shaped:
+          `[max_time, batch_size, cell.output_size]`.
+      output_state_fw is the final state of the forward rnn.
+      output_state_bw is the final state of the backward rnn.
+
+  Raises:
+    TypeError: If `cell_fw` or `cell_bw` is not an instance of `RNNCell`.
+    ValueError: If inputs is None or an empty list.
+  """
+
+  if not isinstance(cell_fw, rnn_cell.RNNCell):
+    raise TypeError("cell_fw must be an instance of RNNCell")
+  if not isinstance(cell_bw, rnn_cell.RNNCell):
+    raise TypeError("cell_bw must be an instance of RNNCell")
+
+  name = scope or "BiRNN"
+  # Forward direction
+  with vs.variable_scope(name + "_FW") as fw_scope:
+    output_fw, output_state_fw = dynamic_rnn(
+        cell_fw, inputs, sequence_length, initial_state_fw, dtype,
+        parallel_iterations, swap_memory, time_major, scope=fw_scope)
+  # Backward direction
+  if not time_major:
+    time_dim = 1
+    batch_dim = 0
+  else:
+    time_dim = 0
+    batch_dim = 1
+  with vs.variable_scope(name + "_BW") as bw_scope:
+    inputs_reverse = array_ops.reverse_sequence(
+        inputs, sequence_length, time_dim, batch_dim)
+    tmp, output_state_bw = dynamic_rnn(
+        cell_bw, inputs_reverse, sequence_length, initial_state_bw, dtype,
+        parallel_iterations, swap_memory, time_major, scope=bw_scope)
+  output_bw = array_ops.reverse_sequence(
+      tmp, sequence_length, time_dim, batch_dim)
+  # Concat each of the forward/backward outputs
+  outputs = array_ops.concat(2, [output_fw, output_bw])
+
+  return (outputs, output_state_fw, output_state_bw)
```

Comment:
perhaps the output state should be a tuple as well?  first one containing fw, second one containing bw?